File: src/go/shipments/retrieve-shipments.go
package main

import (
	"fmt"
	"os"
	"encoding/json"
	"github.com/EasyPost/easypost-go"
)

func main() {
	apiKey := os.Getenv("EASYPOST_API_KEY")
	if apiKey == "" {
		fmt.Fprintln(os.Stderr, "missing API key")
		os.Exit(1)
		return
	}
	client := easypost.New(apiKey)

	// Retrieve a shipments by ID
	shipments, err := client.ListShipments(
		&easypost.ListShipmentsOptions{
			// options here
		},
	)
	if err != nil {
		fmt.Fprintln(os.Stderr, "error retrieving shipments:", err)
		os.Exit(1)
		return
	}

	prettyJSON, err := json.MarshalIndent(shipments, "", "    ")
    if err != nil {
		fmt.Fprintln(os.Stderr, "error creating JSON:", err)
    }
	fmt.Printf("%s\n", string(prettyJSON))
}

File: src/python/shipments/retrieve_shipments.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_PROD_API_KEY')

shipments = easypost.Shipment.all(
    page_size=2,
    start_datetime='2021-01-01T00:00:00Z',
    end_datetime='2021-02-01T00:00:00Z',
)

print(shipments)
